lib/puppet/forge.rb

```diff
@@ -113,8 +113,14 @@ def fetch(input)
     return releases
   end
 
-  def make_http_request(*args)
-    @repository.make_http_request(*args)
+  def make_http_request(uri, *args)
+    # Change plus to space so it can be encoded properly
+    uri = uri.gsub('+', ' ')
```

Comment:
Since this is called by `download` as well, I'm not sure that this is the safest location to make this change.  I think it was intentional to make that mod inline before.